measurements/geography_based.py

```diff
@@ -215,11 +294,76 @@ def geo_tag_gps(dataloader, args):
                 tag_to_region_features[cat][region_name].append((this_features.data.cpu().numpy(), filepath))
         for ann in anns:
             region_tags[region_name][categories.index(ann['label'])] += 1
+            if id_to_subregion is not None:
+                subregion_tags[subregion_name][categories.index(ann['label'])] += 1
     info_stats = {}
     info_stats['region_tags'] = region_tags
+    if id_to_subregion is not None:
+        print("Adding subregion tags...")
+        info_stats['subregion_tags'] = subregion_tags
     info_stats['tag_to_region_features'] = tag_to_region_features
     pickle.dump(info_stats, open("results/{}/geo_tag_gps.pkl".format(args.folder), "wb"))
 
+# private function called from geo_tag() if dataset is of STRING_FORMATTED_LABEL + REGION_LABEL form
+def geo_tag_region(dataloader, args):
+    # map from a region name to a list whose value at index i represents count of category
+    # i 
+    region_tags = {}
+    tag_to_region_features = {}
+    categories = dataloader.dataset.categories
+
+    if not os.path.exists("results/{}/geo_ctr_region.pkl".format(args.folder)):
+        print('running geo_ctr_region() first to get necessary info...')
+        geo_ctr_region(dataloader, args)
+    
+    counts = pickle.load(open("results/{}/geo_ctr_region.pkl".format(args.folder), "rb"))
```

Comment:
same name comment as before about geo_ct_region

Reply:
same, this is a private function, called from geo_ctr()